### PR TITLE
Mjw/dasd 7128 

### PR DESCRIPTION
**Pull request by mwevill**
Initial YAML Build pipeline
Initial ARM template specifying Function app, worker app service and storage account.

### CODE REVIEW

File: azure/template.json [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    },
+    "variables": {
+        "deploymentUrlBase": "https://raw.githubusercontent.com/SkillsFundingAgency/das-platform-building-blocks/master/templates/",
+        "resourceNamePrefix": "[toLower(concat('das-', parameters('EnvironmentName'),'-', parameters('serviceName')))]",
+        "appServicePlanName": "[concat(variables('resourceNamePrefix'),'-asp')]",
+        "functionAppName": "[concat(variables('resourceNamePrefix'),'-fa')]",
+        "storageAccountName": "[toLower(concat('das', parameters('EnvironmentName'), parameters('serviceName'), 'str'))]"
+    },
+    "resources": [
+                {
```
Comment (adam230594) [inline azure/template.json]:
```suggestion
        {
```

Comment (adam230594) [reply azure/template.json]: just the one indent issue but other than that looks gooood

File: azure/template.json [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+            "type": "int",
+            "metadata": {
+                "description": "Number of the app service plan instance"
+            },
+            "defaultValue": 2
+        }
+    },
+    "variables": {
+        "deploymentUrlBase": "https://raw.githubusercontent.com/SkillsFundingAgency/das-platform-building-blocks/master/templates/",
+        "resourceNamePrefix": "[toLower(concat('das-', parameters('EnvironmentName'),'-', parameters('serviceName')))]",
```
Comment (adam230594) [inline azure/template.json]: environmentName

File: azure/template.json [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+        "appServicePlanName": "[concat(variables('resourceNamePrefix'),'-asp')]",
+        "functionAppName": "[concat(variables('resourceNamePrefix'),'-fa')]",
+        "storageAccountName": "[toLower(concat('das', parameters('EnvironmentName'), parameters('serviceName'), 'str'))]"
```
Comment (adam230594) [inline azure/template.json]: environmentName

File: azure/template.json [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+                                "name": "APPINSIGHTS_INSTRUMENTATIONKEY",
+                                "value": "[reference('function-app-insights').outputs.InstrumentationKey.value]"
+                            },
+                            {
+                                "name": "FUNCTIONS_EXTENSION_VERSION",
+                                "value": "~2"
+                            },
+                            {
+                                "name": "EnvironmentName",
+                                "value": "[toUpper(parameters('EnvironmentName'))]"
```
Comment (adam230594) [inline azure/template.json]: environmentName

File: azure/template.json (hunk 2 above)
Comment (mwevill) [reply azure/template.json]: Resolved

File: azure/template.json (hunk 4 above)
Comment (mwevill) [reply azure/template.json]: resolved

**Pull request merged**